doc/source/glossary.rst

```diff
@@ -149,28 +331,51 @@ Glossary
        For more information on dictionaries, read the
        `Python tutorial <https://docs.python.org/tutorial/>`_.
 
+
+   dimension
+
+       See :term:`axis`.
+
+
+   dtype
+
+       The datatype describing the (identically typed) elements in an ndarray.
+       It can be changed to reinterpret the array contents. For details, see
+       :doc:`Data type objects (dtype). <reference/arrays.dtypes>`
+
+
+   fancy indexing
+
+       Another term for :term:`advanced indexing`.
+
+
    field
        In a :term:`structured data type`, each sub-type is called a `field`.
        The `field` has a name (a string), a type (any valid dtype), and
        an optional `title`. See :ref:`arrays.dtypes`
 
+
    Fortran order
        See `column-major`
 
+
    flattened
        Collapsed to a one-dimensional array. See `numpy.ndarray.flatten`
        for details.
 
+
    homogeneous
-       Describes a block of memory comprised of blocks, each block comprised of 
+       Describes a block of memory comprised of blocks, each block comprised of
        items and of the same size, and blocks are interpreted in exactly the
        same way. In the simplest case each block contains a single item, for
        instance int32 or float64.
 
+
    immutable
        An object that cannot be modified after execution is called
        immutable.  Two common examples are strings and tuples.
 
+
    iterable
```

Comment:
Happy to declare this out of scope for this PR, but `` :term:`python:iterable` `` exists.

Reply:
Agreed, doesn't belong here. Immutable and iterable are entries I didn't want in the NumPy glossary because they're Python glossary terms. But as @mattip notes above, they're preexisting entries -- not '+'.